Tribler/Core/CacheDB/SqliteCacheDBHandler.py

```diff
@@ -515,6 +515,20 @@ def updateTorrent(self, infohash, notify=True, **kw):  # watch the schema of dat
         if notify:
             self.notifier.notify(NTFY_TORRENTS, NTFY_UPDATE, infohash)
 
+    def update_torrent_with_metainfo(self, infohash, metainfo):
+        """ Updates name, length and num files from metainfo. """
```

Comment:
Shouldn't you try to load this from the database (if it exists)? You run the risk of overwriting known data now.

Reply:
Check from the database is done by the caller (torrentinfo_endpoint.py:69). But I agree with your comment.

Reply:
Yeah, it would be cleaner if the `torrentinfo_endpoint.py` does not concern itself with the existing database state. That should be in this function